source/device_defender.c

```diff
@@ -1365,6 +1410,74 @@ int aws_iotdevice_defender_config_register_ip_list_metric(
     return AWS_OP_ERR;
 }
 
+int aws_iotdevice_defender_config_register_number_double_metric(
+    struct aws_iotdevice_defender_task_config *task_config,
+    const struct aws_byte_cursor *metric_name,
+    aws_iotdevice_defender_get_number_double_fn *supplier,
+    void *userdata) {
+    AWS_PRECONDITION(task_config != NULL);
+    AWS_PRECONDITION(metric_name != NULL);
+    AWS_PRECONDITION(supplier != NULL);
+
+    struct aws_allocator *allocator = task_config->allocator;
+    struct defender_custom_metric *custom_metric = aws_mem_calloc(allocator, 1, sizeof(struct defender_custom_metric));
+    if (custom_metric) {
```

Comment:
memory allocation can't fail anymore, don't need this `if` check. In the past we didn't guarantee this, so you'll see older code doing the checks. But newer code can be simpler